### ISSUE TITLE
Crash during recording just after opening a project.

### ISSUE THREAD

**Issue by AdTb** [monocasual/giada#301]
OS: MacOS High Sierra v10.13.6
Giada Version 0.16.1 ( Jan 6 2020)

Minimum reproducible bug (looks similar to issue #161) :

1.  Start a new project

2. Set the beats to 16 and keep the bar to 1
3. Save the project (foo.gprj)
4. Close and reopen Giada (it is very important, to be sure that it does not have more memory allocated than necessary)
5. Open the project foo.gprj
6. Add a sample channel to a column and arm it for recording
7. Start recording
8. After recording during 4 beats, the software crashes.

The MAC crash report provides the following information about the error
````
Exception Type:        EXC_BAD_ACCESS (SIGBUS)
Exception Codes:       KERN_PROTECTION_FAILURE at 0x00000001046f4000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Bus error: 10
Termination Reason:    Namespace SIGNAL, Code 0xa
Terminating Process:   exc handler [0]
````

When you set again the beats to 16 between (without actually changing anything to the config) the steps 5 and 6, Giada does not crash.

My interpretation of the bug: there is might a variable that defines how much memory to allocate to a new recording. When the beats are set from 4 to 16, this variable is increased to match the actual size. When you load the existing project `foo.gprj`, this variable is not well initialized and is set to 4 (the default) instead of 16. When you try to record a new sample, the memory allocated will only be enough for 4 beats and thus, after these 4 beats a bad memory access occurs, causing the crash. 
I am not a hundred percent sure of that (I have not looked at the code yet), but it seems probable.

**Comment by AdTb** [comment]
Ok, I found the issue, the segfault is due to a buffer overflow at line 148 of `mixer.cpp`, because the AudioBuffer `vChanInput_` has not enough memory allocated.
This is due to the loading patch process, around the line 196 of `glue/storage.cpp`. The AudioBuffer is first initialized with a clock having a default of 4 beats, then the clock is updated to 16 beats, but the buffer is not updated (contrary to the "user driven" updating of a beat at line 147 of `glue/main.cpp`). 
I made a pull request with a change that fixes it (and the tests are ok).
PS: It is a pleasure to read a code so well written,  that's a good job.
